network/limitcaller/rateLimitingTransport.go

```diff
@@ -59,37 +59,36 @@ func MakeRateLimitingTransport(phonebook ConnectionTimeStore, queueingTimeout ti
 		ExpectContinueTimeout: defaultTransport.ExpectContinueTimeout,
 		MaxIdleConnsPerHost:   maxIdleConnsPerHost,
 	}
-	return MakeRateLimitingTransportWithRoundTripper(phonebook, queueingTimeout, innerTransport, nil, maxIdleConnsPerHost)
+	return MakeRateLimitingBoundTransportWithRoundTripper(phonebook, queueingTimeout, innerTransport, target)
 }
 
-// MakeRateLimitingTransportWithRoundTripper creates a rate limiting http transport that would limit the requests rate
-// according to the entries in the phonebook.
-func MakeRateLimitingTransportWithRoundTripper(phonebook ConnectionTimeStore, queueingTimeout time.Duration, rt http.RoundTripper, target interface{}, maxIdleConnsPerHost int) RateLimitingTransport {
-	return RateLimitingTransport{
+// MakeRateLimitingBoundTransportWithRoundTripper creates a rate limiting http transport that:
+// 1. would limit the requests rate according to the entries in the phonebook.
+// 2. is bound to a specific target.
+func MakeRateLimitingBoundTransportWithRoundTripper(phonebook ConnectionTimeStore, queueingTimeout time.Duration, rt http.RoundTripper, target string) RateLimitingBoundTransport {
+	return RateLimitingBoundTransport{
 		phonebook:       phonebook,
 		innerTransport:  rt,
 		queueingTimeout: queueingTimeout,
-		targetAddr:      target,
+		addrOrPeerID:    target,
 	}
 }
 
 // RoundTrip connects to the address on the named network using the provided context.
 // It waits if needed not to exceed connectionsRateLimitingCount.
-func (r *RateLimitingTransport) RoundTrip(req *http.Request) (res *http.Response, err error) {
+func (r *RateLimitingBoundTransport) RoundTrip(req *http.Request) (res *http.Response, err error) {
 	var waitTime time.Duration
 	var provisionalTime time.Time
-	queueingDeadline := time.Now().Add(r.queueingTimeout)
-	addrOrPeerID := req.Host
-	// p2p/http clients have per-connection transport and address info so use that
-	if len(req.Host) == 0 && req.URL != nil && len(req.URL.Host) == 0 {
-		addrInfo, ok := r.targetAddr.(*peer.AddrInfo)
-		if !ok {
-			return nil, errors.New("rateLimitingTransport: request without Host/URL and targetAddr is not a peer.AddrInfo")
-		}
-		addrOrPeerID = string(addrInfo.ID)
+	if r.addrOrPeerID == "" {
+		return nil, errors.New("rateLimitingTransport: target not set")
+	}
+	if req.URL != nil && req.URL.Host != "" && req.URL.Host != r.addrOrPeerID {
```

Comment:
This is basically saying "if there is an http address target, it has to match the request url".

Reply:
yes since it is "bound" limiter and transport